ocf_data_sampler/torch_datasets/datasets/pvnet_uk.py

```diff
@@ -186,9 +214,11 @@ def __init__(
         """
         
         config = load_yaml_configuration(config_filename)
-        
```

Comment:
Could you do it directly after loading the config, and not use dataset_dict. 
Inside the function you can use `config.input_data.satellite.channels` and similar for nwp. 
This would mean it would catch the error earlier, and is independent from `dataset_dict`